### PR TITLE
microsocks: init at 1.0.4

### PR DESCRIPTION
**Pull request by Ramblurr**
## Description of changes

microsocks is a tiny, portable SOCKS5 server with very moderate resource usage.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>microsocks</li>
  </ul>
</details>

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/mi/microsocks/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/mi/microsocks/package.nix:14]: It's preferred to write `hash` when specifying the hash value.

```suggestion
    hash = "sha256-cB2XMWjoZ1zLAmAfl/nqjdOyBDKZ+xtlEmqsZxjnFn0=";
```

File: pkgs/by-name/mi/microsocks/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/mi/microsocks/package.nix:21]: Please add the `meta.mainProgram` tag.

File: pkgs/by-name/mi/microsocks/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/mi/microsocks/package.nix:23]: See: [https://github.com/NixOS/nixpkgs/blob/3551c607f454ee855e54db5686464c8dcf83ceae/doc/stdenv/meta.chapter.md#description-var-meta-description](https://github.com/NixOS/nixpkgs/blob/3551c607f454ee855e54db5686464c8dcf83ceae/doc/stdenv/meta.chapter.md#description-var-meta-description)

```suggestion
    description = "Tiny, portable SOCKS5 server with very moderate resource usage";
```

**Review by MikaelFangel — COMMENTED** [review]
Just added some quick comments (: Thanks for your contribution.

And please rename this PR to "microsocks: init at version 1.0.4"

**Comment by Ramblurr** [comment]
Thanks for the suggestions @MikaelFangel . Should I force push to my branch to keep it at 2 commits?

**Comment by MikaelFangel** [comment]
That is correct (:

**Comment by Ramblurr** [comment]
All fixed up!

**Comment by MikaelFangel** [comment]
Nice! Missed that you should remove "version" from both the commit and the PR title. (:

**Comment by Ramblurr** [comment]
> Nice! Missed that you should remove "version" from both the commit and the PR title. (:

fixed :)

**Review by MikaelFangel — APPROVED** [review]
Result of `nixpkgs-review pr 296852` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>microsocks</li>
  </ul>
</details>

**Comment by Ramblurr** [comment]
@MikaelFangel I have some other new packages I'd like to open PRs for, but this PR has the commit that adds me as a maintainer.. how should I go about handling that?

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:21]:
```suggestion
  meta = {
```

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:26]:
```suggestion
    maintainers = with lib.maintainers; [ ramblurr ];
```

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:24]:
```suggestion
    license = lib.licenses.mit;
```

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:9]:
```suggestion
  version = "1.0.4";

  src = fetchFromGitHub {
```

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:17]: the install hooks are missing.

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:27]:
```suggestion
```
Default value.

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:13]:
```suggestion
    fetchSubmodules = true;
```
What happens if you remove this?

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:6]:
```suggestion
{
  stdenv,
  fetchFromGitHub,
  lib
}:

stdenv.mkDerivation rec {
```

It's usual to skip a line after parameters.

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [reply pkgs/by-name/mi/microsocks/package.nix:21]: `microsocks` is default value because is equal to pname.
Why to add `meta.mainProgram`?

**Comment by superherointj** [comment]
Always squash commits. It should be a commit per package/module/etc.

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:22]:
```suggestion
  meta = {
    changelog = "https://github.com/rofl0r/microsocks/releases/tag/v${version}";
    description = "Tiny, portable SOCKS5 server with very moderate resource usage";
```

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [reply pkgs/by-name/mi/microsocks/package.nix:17]: Still missing. Example:
>   installPhase = ''
>     runHook preInstall
>     # Content goes here
>     runHook postInstall
>   '';

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:19]:
```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm 755 microsocks -t $out/bin/
    runHook postInstall
  '';
```

File: pkgs/by-name/mi/microsocks/package.nix
Comment (Ramblurr) [reply pkgs/by-name/mi/microsocks/package.nix:17]: good catch thanks!

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:6]:
```suggestion
{ stdenv,
  fetchFromGitHub,
  lib,
}:

stdenv.mkDerivation rec {
```

Comment (Ramblurr) [reply pkgs/by-name/mi/microsocks/package.nix:6]: Ah.. my alejandra formatter doesn't  like that extra line break.

**Comment by Ramblurr** [comment]
@superherointj  I have a nixos module for microsocks.. should I include it in this PR or make a new one after this is merged?

**Comment by superherointj** [comment]
Feel free to include in this PR, but it has to be a different commit.

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:32]:
```suggestion
    maintainers = with lib.maintainers; [ ramblurr ];
```

File: pkgs/by-name/mi/microsocks/package.nix
Comment (superherointj) [inline pkgs/by-name/mi/microsocks/package.nix:2]:
```suggestion
{ stdenv,
```

File: nixos/modules/services/networking/microsocks.nix
Comment (superherointj) [inline nixos/modules/services/networking/microsocks.nix:7]:
```suggestion
let
```

File: nixos/modules/services/networking/microsocks.nix
Comment (superherointj) [inline nixos/modules/services/networking/microsocks.nix:109]:
```suggestion
      after = [ "network.target" ];
```

File: nixos/modules/services/networking/microsocks.nix
Comment (superherointj) [inline nixos/modules/services/networking/microsocks.nix:110]:
```suggestion
      wantedBy = [ "multi-user.target" ];
```

File: nixos/modules/services/networking/microsocks.nix
Comment (superherointj) [inline nixos/modules/services/networking/microsocks.nix:127]:
```suggestion
        RestrictNamespaces = [ "uts" "ipc" "pid" "user" "cgroup" ];
```
As always, add spaces in `[` and `]`.
Also whenever possible, default to alphabetically ordering items in lists and attrs.
Exception is, `meta.broken` is always last because to avoid git conflicts. (It's a field that changes more often.)
I wish we had a linter, a formatter, that would get things right. No kidding. It's very debatable. And I having to give back such feedback. I'm just parroting what I was told when I first got here.

File: nixos/modules/services/networking/microsocks.nix
Comment (superherointj) [reply nixos/modules/services/networking/microsocks.nix:7]: Avoid `with` like the plague.

File: nixos/modules/services/networking/microsocks.nix
Comment (Ramblurr) [reply nixos/modules/services/networking/microsocks.nix:127]: A formatter would be nice. All the nits you've identified are table stakes for any formatter.

It's quite a shame that [Alejandra](https://github.com/kamadorueda/alejandra)
isn't formatting how you'd like since it seems to be the most common formatter. 

Anyways I'll fix them up.

Comment (superherointj) [reply nixos/modules/services/networking/microsocks.nix]: Let's settle for Alejandra.

**Review by superherointj — APPROVED** [review]

File: nixos/modules/services/networking/microsocks.nix
Comment (superherointj) [reply nixos/modules/services/networking/microsocks.nix]: There is also: https://hackage.haskell.org/package/nixfmt

https://github.com/NixOS/rfcs/pull/166

**Pull request merged**
